### PR TITLE
Fix new lint errors from pylint 1.8.1.

### LINKED ISSUE #553: Lint errors with pylint 1.8.1 update
New pylint version has more things for us to improve in our code!

### PR DESCRIPTION
**Pull request by noahl**
Previously we were linting with pylint 1.7.5. The new release adds
some new checks.

Closes #553 .

### CODE REVIEW

File: rho/utilities.py [hunk 1]
```diff
@@ -181,16 +181,19 @@ def tail_log(path, ansible_verbosity, process_output, env=None):
...
-        if env is not None and isinstance(env, dict):
-            for key, value in iteritems(env):
-                os.environ[key] = str(value)
-        # pylint: disable=no-member
-        process = sh.tail('-f', '-n', '+0', path, _out=process_output,
-                          _bg=True, _bg_exc=False)
-        return process
+
+    if not path:
+        raise ValueError('tail_log must have a nonempty path for logs')
```
Comment (chambridge) [inline rho/utilities.py:186]: If we are going to make this change to raise an error we need to update the `run_with_vault()` method to appropriately catch the error and log it for the user:
https://github.com/quipucords/rho/blob/master/rho/ansible_utils.py#L142

**Review by chambridge — COMMENTED** [review]
Comment on the error handling.

Can we also lock the dev-requirement.txt for this?

File: rho/utilities.py (hunk 1 above)
Comment (noahl) [reply rho/utilities.py:186]: Done

**Comment by noahl** [comment]
@chambridge I just pushed a new commit that I think fixes all of the points you made; please take a look.

I locked the version of `pylint`, and also `flake8` because the same thing could happen with that.

**Review by chambridge — APPROVED** [review]

**Pull request merged**